lib/checks/keyboard/main-is-top-level.js

```diff
@@ -0,0 +1,13 @@
+var landmarks = axe.commons.aria.getRolesByType('landmark');
+var parent = node.parentElement;
+while (parent){
+	var role = parent.getAttribute('role');
+	if (!role){
+		role = axe.commons.aria.implicitRole(parent);
```

Comment:
@jongund Both of those suggestions make sense to me. Seeing as it's still an open discussion if forms should be a landmark, make an exception for this rule. That one is particularly relevant for the `region` rule. You can adjust the type of application and presentation in `lib/commons/aria/index`.